snapcraft/plugins/roscore.py

```diff
@@ -67,6 +68,8 @@ def build(self):
 
         os.chmod(service_wrapper, 0o755)
 
+        self._finish_build()
```

Comment:
Personally, I would find it more readable if you called that smattering of things one by one instead of grouping them in a function that will have more than one responsibility and with a name that will force you to read the code to understand what it's doing.
But I'm just leaving this as a suggestion, I'm not blocking your PR on this. I understand why you have to patch things afterwards, and I'm not expecting that to be pretty at all.